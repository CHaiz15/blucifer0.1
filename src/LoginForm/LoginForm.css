.main-content {
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 100vh;
  background-image: url("../assets/bluciferHERO.jpg");
  background-repeat: no-repeat;
  background-size: auto;
  overflow: hidden;
}

.title {
  -webkit-text-stroke-width: 2px;
  -webkit-text-stroke-color: #FB4F14;
  margin: 2% 0 0 0;
  font-weight: 900;
  font-size: 2.5em;
  color: #ffffff;
  text-shadow: 0 0 3px #000;
}

.text-center {
	color:#fff;	
  font-size: 1.5em;
  margin: 0 0 20% 0;
  display: flex;
  justify-content: center;
}

.login-form {
  display: flex;
  flex-direction: column;
  background-color: rgba(0, 0, 0, 0.75);
	border-radius: 3px;
  padding: 5% 6%;
  margin: 2% 0 0 0;
  box-shadow: 0 0 3px #000;
}

.input-container {
	position: relative;
  margin-bottom: 25px;
  font-weight: bolder;
}

.input-container label{
	position: absolute;
	top: 0px;
	left: 0px;
	font-size: 16px;
	color:#fff;	
	transition: all 0.5s ease-in-out;
}

.input-container input{ 
  border: 0;
  border-bottom: 1px solid #555;  
  background: transparent;
  width: 100%;
  padding: 6% 0 0.5% 0;
  font-size: 20px;
  color:#fff;
}

.input-container input:focus { 
  border: none;	
  outline: none;
  border-bottom: 1px solid #FB4F14;	
}

.input-container input:focus ~ label,
.input-container input:valid ~ label{
  margin: -2% 0 0 0;
	font-size:12px;
}

.purpose-buttons {
  margin: -10% 0 0 0;
  display: flex;
  justify-content: space-around;
}

.purpose-btn {
	color:#fff;
	background-color:#002244;
	outline: none;
  border: 0;
  color: #fff;
	padding: 10px 20px;
	margin-top: 50px;
	border-radius: 2px;
	cursor: pointer;
  position: relative;
  width: 33.3%;
  text-align: center;
}

.explore-btn {
	color:#fff;
	background-color :#FB4F14;
	outline: none;
  border: 0;
  color: #fff;
	padding: 10px 20px;
  margin-top: 50px;
  margin-bottom: 10px;
	border-radius: 2px;
	cursor: pointer;
  position: relative;
  width: 100%;
}

.purpose-btn:focus,
.purpose-btn:hover,
.explore-btn:hover {
  box-shadow: inset 0px 0px 3px #ffffff;
}

.hidden {
  visibility: hidden;
}

.input-error {
  text-align: center;
  color: #FF9494;
  animation: shake-top 0.8s cubic-bezier(0.455, 0.030, 0.515, 0.955) both;
}

@keyframes shake-top {
  0%,
  100% {
    transform: rotate(0deg);
    transform-origin: 50% 0;
  }
  10% {
    transform: rotate(0deg);
  }
  20%,
  40%,
  60% {
    transform: rotate(-1deg);
  }
  30%,
  50%,
  70% {
    transform: rotate(1deg);
  }
  80% {
    transform: rotate(1deg);
  }
  90% {
    transform: rotate(-1deg);
  }
}
