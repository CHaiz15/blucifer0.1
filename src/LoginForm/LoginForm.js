import React, { Component } from 'react'
import './LoginForm.css'

class LoginForm extends Component {
  constructor() {
    super();

    this.state = {
      name: '',
      purpose: ''
    }
  }
  updateChange = (e) => {
    e.preventDefault();
    this.setState({[e.target.name]: e.target.value})
  }

  submitForm = (e) => {
    e.preventDefault()
    this.props.addFormInfo(this.state)
  }

  render() {
    return (
      <form className='login-form'>
        <label>Name</label>
        <input
          name='name'
          placeholder='Name...'
          value={this.state.name}
          onChange={this.updateChange}
        />
        <label>Email</label>
        <input
          placeholder='Email...'
        />
        <div>
          <button
            name='purpose'
            value='Business'
            onClick={this.updateChange}
          >Business</button>
          <button
            name='purpose'
            value='Vacation'
            onClick={this.updateChange}
          >Vacation</button>
          <button
            name='purpose'
            value='Travel'
            onClick={this.updateChange}
          >Other</button>
        </div>
        <button onClick={this.submitForm}>Explore</button>
      </form>
    )
  }
}

export default LoginForm;
