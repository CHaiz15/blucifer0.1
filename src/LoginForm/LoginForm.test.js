import React from 'react';
import { render } from '@testing-library/react';
import LoginForm from './LoginForm';
import { shallow } from 'enzyme';

describe('LoginForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoginForm/>);
  })

  describe('Default state', () => {
  
    it('should be an instance of LoginForm component', () => {
        expect(wrapper).toMatchSnapshot();
    });
  
    it('should start state with default name', () => {
      expect(wrapper.state('name')).toEqual('')
    })
  
    it('should start state with default email', () => {
      expect(wrapper.state('email')).toEqual('')
    })
  
    it('should start state with default purpose', () => {
      expect(wrapper.state('purpose')).toEqual('') 
    })
  
    it('should start state with default showError', () => {
      expect(wrapper.state('showError')).toEqual(false) 
    })
  
    it('should start state with default formFilled', () => {
      expect(wrapper.state('formFilled')).toEqual(false)
    })
  })

  describe('updateChange', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<LoginForm />);
    })
    it('should change state of name if form is updated', () => {
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'name',
          value: 'Kim'
        }
      }

      wrapper.instance().updateChange(mockEvent)
      expect(wrapper.state('name')).toEqual('Kim')
    })

    it('should change state of name if form is updated', () => {
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'email',
          value: 'kim@kim.com'
        }
      }

      wrapper.instance().updateChange(mockEvent)
      expect(wrapper.state('email')).toEqual('kim@kim.com')
    })

    it('should change state of name if form is updated', () => {
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'purpose',
          value: 'business'
        }
      }

      wrapper.instance().updateChange(mockEvent)
      expect(wrapper.state('purpose')).toEqual('business')
    })
  })

  describe('submitForm', () => {
    let wrapper;
    const mockAddFormInfo = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<LoginForm addFormInfo={mockAddFormInfo} />);
    })
    it('should change state of formFilled if form is filled', () => {
      const mockEvent = {preventDefault: jest.fn()}
      wrapper.setState({
        name: 'Kim',
        email: 'kim@kim.com',
        purpose: 'business'
      })
      wrapper.instance().submitForm(mockEvent)
      expect(wrapper.state('formFilled')).toEqual(true)
    })

    it('should change state of showError if form is not filled', () => {
      const mockEvent = {preventDefault: jest.fn()}
      wrapper.setState({
        name: '',
        email: '',
        purpose: 'business'
      })
      wrapper.instance().submitForm(mockEvent)
      expect(wrapper.state('showError')).toEqual(true)
    })
  })
})