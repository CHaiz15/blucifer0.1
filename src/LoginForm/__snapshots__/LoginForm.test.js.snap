// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Default state should be an instance of LoginForm component 1`] = `
<main
  className="main-content"
>
  <h1
    className="title"
  >
    BLUCIFER BNB
  </h1>
  <form
    className="login-form"
  >
    <h3
      className="text-center"
    >
      LOGIN
    </h3>
    <div
      className="input-container"
    >
      <input
        autoComplete="off"
        className="user-input"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label>
        Full Name
      </label>
    </div>
    <div
      className="input-container"
    >
      <input
        autoComplete="off"
        className="user-input"
        name="email"
        onChange={[Function]}
        required={true}
        type="mail"
        value=""
      />
      <label>
        Email
      </label>
    </div>
    <div
      className="purpose-buttons"
    >
      <button
        className="purpose-btn"
        name="purpose"
        onClick={[Function]}
        value="Business"
      >
        BUSINESS
      </button>
      <button
        className="purpose-btn"
        name="purpose"
        onClick={[Function]}
        value="Vacation"
      >
        VACATION
      </button>
      <button
        className="purpose-btn"
        name="purpose"
        onClick={[Function]}
        value="Travel"
      >
        OTHER
      </button>
    </div>
    <button
      className="explore-btn"
      onClick={[Function]}
      type="button"
    >
      EXPLORE DENVER
    </button>
    <h4
      className="hidden"
    >
      Please fill all inputs.
    </h4>
  </form>
</main>
`;
